harness/determined/pytorch/deepspeed/_deepspeed_context.py

```diff
@@ -0,0 +1,386 @@
+import contextlib
+import importlib.util
+import json
+import logging
+import os
+from typing import Any, Dict, Iterator, List, Optional, Set, Tuple, Type, Union, cast
+
+import torch
+import torch.nn as nn
+from deepspeed import PipelineEngine
+from deepspeed.runtime.config_utils import dict_raise_error_on_duplicate_keys
+
+import determined as det
+from determined import profiler, pytorch
+from determined.common import check
```

Comment:
don't even import the check module in new code.

For validating user inputs, use if statements, otherwise use asserts.